packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1594,9 +1605,17 @@ abstract class FlutterCommand extends Command<void> {
     FlutterCommandResult commandResult,
     DateTime startTime,
     DateTime endTime,
+    Analytics analytics,
+    ProcessInfo processInfo,
   ) {
```

Comment:
@christopherfujino, is it enough to inject it at this level? Otherwise, I would need to inject these 2 parameters in each of the 70+ `FlutterCommand` implementations.

Reply:
or I could possibly create a getter in the abstract class `FlutterCommand` to fetch these two classes from globals so that each implementation of `FlutterCommand` has access to it

```dart
abstract class FlutterCommand extends Command<void> {
  
  Analytics get analytics => globals.analytics;
  ProcessInfo get processInfo => globals.processInfo;
  
  //.....
}

Reply:
you know what, I like that approach. Then later on, we can set up a method to lazily initialize them.

Reply:
Cool, that will also make it easy for any other command to directly inject the analytics instance too

Reply:
And for posterity, when I saw "a method to lazily initialize them", I mean a process, not literally a class method.